server/src-lib/Hasura/GraphQL/Context.hs

```diff
@@ -116,3 +139,51 @@ type MutationRootField v =
 
 type SubscriptionRootField v = RootField (QueryDB 'Postgres v) Void (RQL.AnnActionAsyncQuery 'Postgres v) Void
 type SubscriptionRootFieldResolved = RootField (QueryDB 'Postgres S.SQLExp) Void (RQL.AnnSimpleSel 'Postgres) Void
+
+resolveRemoteVariable
+  :: (MonadError RQL.QErr m)
+  => UserInfo
+  -> RQL.RemoteSchemaVariable
+  -> m Variable
+resolveRemoteVariable userInfo = \case
+  RQL.SessionPresetVariable sessionVar gType varName presetType -> do
+    sessionVarVal <- onNothing (getSessionVariableValue sessionVar $ _uiSession userInfo)
+      $ RQL.throw400 RQL.NotFound $ sessionVar <<> " session variable expected, but not found"
+    let baseType = G.getBaseType gType
```

Comment:
That's a good point! I had thought about this while writing this part of the code. I thought that maybe since the argument is a "preset" we should be throwing an error when we don't get the value from the session variable. I'm still not sure what's the right behaviour. 

cc: @tirumaraiselvan

Reply:
All our presets so far (in tables) have been _hard_ presets. In case a session variable is expected in permission, then we throw an error if it's not given even if it is technically "nullable". I think we should keep the same behaviour.